vignettes/irecist.Rmd

```diff
@@ -0,0 +1,877 @@
+  condition = AVALC == "iCR" &
+    all(AVALC.join %in% c("iCR", "NE")) &
+    count_vals(var = AVALC.join, val = "NE") <= 1,
+  set_values_to = exprs(AVALC = "iCR")
+)
+
+icbor_ipr <- event_joined(
+  description = paste(
+    "Define partial response (iPR) for confirmed best overall response (iCBOR) as",
+    "iPR followed by iCR or iPR at least 28 days later, at most one NE in between,",
+    "and no iPR after iCR"
+  ),
+  dataset_name = "ovr",
+  join_vars = exprs(AVALC, ADT),
+  join_type = "after",
+  first_cond_upper = AVALC.join %in% c("iCR", "iPR") &
+    ADT.join >= ADT + confirmation_period,
+  condition = AVALC == "iPR" &
+    all(AVALC.join %in% c("iCR", "iPR", "NE")) &
+    count_vals(var = AVALC.join, val = "NE") <= 1 &
+    (
+      min_cond(
+        var = ADT.join,
+        cond = AVALC.join == "iCR"
+      ) > max_cond(var = ADT.join, cond = AVALC.join == "iPR") |
+        count_vals(var = AVALC.join, val = "iCR") == 0 |
+        count_vals(var = AVALC.join, val = "iPR") == 0
+    ),
+  set_values_to = exprs(AVALC = "iPR")
+)
+```
+
+Please note that the result `AVALC = "Y"` for confirmed clinical benefit is
+defined by the first _two_ events specified for `events`. For subjects with
+observations fulfilling both events the one with the earlier date should be
+selected (and not the first one in the list).
+
+```{r}
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(desc(AVALC), ADT),
+    mode = "first",
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    events = list(icrsp_y_icr, icrsp_y_ipr, no_data_n),
+    set_values_to = exprs(
+      PARAMCD = "ICRSP",
+      PARAM = "Immune Confirmed Response by Investigator",
+      PARCAT1 = "Tumor Response",
+      PARCAT2 = "Investigator",
+      PARCAT3 = "iRECIST",
+      AVAL = yn_to_numeric(AVALC),
+      ANL01FL = "Y"
+    )
+  )
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(desc(AVALC), ADT),
+    mode = "first",
+    events = list(icrsp_y_icr, icrsp_y_ipr, icb_y, no_data_n),
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    set_values_to = exprs(
+      PARAMCD = "ICCB",
+      PARAM = "Immune Confirmed Clinical Benefit by Investigator",
+      PARCAT1 = "Tumor Response",
+      PARCAT2 = "Investigator",
+      PARCAT3 = "iRECIST",
+      AVAL = yn_to_numeric(AVALC),
+      ANL01FL = "Y"
+    ),
+    check_type = "none"
+  )
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    tmp_event_nr_var = event_nr,
+    order = exprs(event_nr,ADT),
+    mode = "first",
+    events = list(icbor_icr, icbor_ipr, ibor_isd, ibor_non_icriupd, ibor_iupd, ibor_icpd, ibor_ne, no_data_missing),
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    set_values_to = exprs(
+      PARAMCD = "ICBOR",
+      PARAM = "Immune Best Confirmed Overall Response by Investigator",
```

Comment:
General note: I am not sure that the prefix "Immune" for PARAM variables is accurate. Maybe instead of the "Immune" prefix use "iRecist"?